collect_app/src/test/java/org/odk/collect/android/storage/migration/StorageMigratorTest.java

```diff
@@ -0,0 +1,163 @@
+package org.odk.collect.android.storage.migration;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.odk.collect.android.storage.StoragePathProvider;
+import org.odk.collect.android.storage.StorageStateProvider;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class StorageMigratorTest {
+
+    private StorageMigrator storageMigrator;
+    private StorageEraser storageEraser;
+    private final StoragePathProvider storagePathProvider = spy(StoragePathProvider.class);
+    private final StorageStateProvider storageStateProvider = mock(StorageStateProvider.class);
+
+    @Before
+    public void setup() {
+        storageEraser = spy(new StorageEraser(storagePathProvider));
```

Comment:
Again would be nice for this to be a mock/fake rather than a spy.

Reply:
Fixed.